core/domain/collection_services.py

```diff
@@ -1290,14 +1256,8 @@ def index_collections_given_ids(collection_ids):
         collection_ids: list(str). List of collection ids whose collections are
             to be indexed.
     """
-    # We pass 'strict=False' so as not to index deleted collections.
-    collection_list = get_multiple_collections_by_id(
-        collection_ids, strict=False).values()
-    search_services.add_documents_to_index([
-        _collection_to_search_dict(collection)
-        for collection in collection_list
-        if _should_index(collection)
-    ], SEARCH_INDEX_COLLECTIONS)
+    collection_summaries = get_collection_summaries_matching_ids(collection_ids)
+    search_services.index_collection_summaries(collection_summaries)
```

Comment:
Should probably filter the list to remove "None" results, right, since the previous function can return None?

Reply:
Done.